rest/src/main/java/com/redhat/ipaas/rest/v1/dao/ConnectionDAO.java

```diff
@@ -15,26 +15,30 @@
  */
 package com.redhat.ipaas.rest.v1.dao;
 
+import com.redhat.ipaas.rest.v1.controller.handler.rtdb.RealTimeDB;
+import com.redhat.ipaas.rest.v1.model.connection.Connection;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.stereotype.Component;
+import org.springframework.stereotype.Service;
 
-import java.util.Arrays;
-import java.util.List;
+import javax.inject.Named;
 
-@Component
-public class DefaultDataAccessObjectProvider implements DataAccessObjectProvider {
-
-    private final IntegrationDAO integrationDAO;
-
-    private final IntegrationPatternDAO integrationPatternDAO;
+@Named(Connection.KIND)
+@Service
+public class ConnectionDAO extends RealTimeAbstractDAO<Connection> {
```

Comment:
The REST API needs something simpler.  The ui wants to work with an integration as a whole json doc.  It does not want to have to join it in the UI.  How or IF we normalize data is up to the DAO layer.